modules/tensor_mechanics/test/tests/substepping/tests

```diff
@@ -5,45 +5,52 @@
-               'Outputs/file_base=ad_power_law_out'
-    requirement = 'The system shall report diverged return mapping solve due to large deformation '
-                  'imposed in one step'
+    cli_args = 'Materials/power_law_creep/use_substepping=NONE Executioner/dtmin=0.1 Outputs/file_base=ad_power_law_out'
+    requirement = 'The system shall report diverged return mapping solve due to large deformation imposed in one step'
   []
   [ad_power_law_substepping]
     type = 'Exodiff'
     input = 'ad_power_law_creep.i'
-    cli_args = 'Materials/power_law_creep/use_substep=true '
-               'Materials/power_law_creep/maximum_number_substeps=500 Executioner/dtmin=0.1 '
-               'Outputs/file_base=ad_power_law_substepping_out'
+    cli_args = "Materials/power_law_creep/use_substepping=INCREMENT_BASED "
+               "Materials/power_law_creep/maximum_number_substeps=500 Executioner/dtmin=0.1 "
+               "Outputs/file_base=ad_power_law_substepping_out"
     exodiff = 'ad_power_law_substepping_out.e'
     valgrind = HEAVY
-    requirement = 'The system shall converge under large deformation owing to the help of '
-                  'substepping'
+    requirement = 'The system shall converge under large deformation owing to the help of substepping'
   []
   [power_law]
     type = 'RunException'
     input = 'power_law_creep.i'
     expect_err = 'Exceeded maximum iterations in material return mapping iterations.'
-    cli_args = 'Materials/power_law_creep/use_substep=false Executioner/dtmin=0.1 '
-               'Outputs/file_base=power_law_out'
-    requirement = 'The system shall report diverged return mapping solve due to large deformation '
-                  'imposed in one step'
+    cli_args = 'Executioner/dtmin=0.1 Outputs/file_base=power_law_out'
+    requirement = 'The system shall report diverged return mapping solve due to large deformation imposed in one step'
   []
   [power_law_substepping]
     type = 'Exodiff'
     input = 'power_law_creep.i'
-    cli_args = 'Materials/power_law_creep/use_substep=true '
-               'Materials/power_law_creep/maximum_number_substeps=500 Executioner/dtmin=0.1 '
-               'Outputs/file_base=power_law_substepping_out'
+    cli_args = "Materials/power_law_creep/use_substepping=INCREMENT_BASED "
+               "Materials/power_law_creep/maximum_number_substeps=500 Executioner/dtmin=0.1 "
+               "Outputs/file_base=power_law_substepping_out"
     exodiff = 'power_law_substepping_out.e'
-    requirement = 'The system shall converge under large deformation owing to the help of '
-                  'substepping'
+    requirement = 'The system shall converge under large deformation owing to the help of substepping'
+  []
+  [power_law_adaptive_substepping]
+    # We only check that the simulation converges with adaptive substepping
+    type = 'RunApp'
+    input = 'power_law_creep.i'
+    cli_args = "Materials/power_law_creep/use_substepping=INCREMENT_BASED "
```

Comment:
No `ERROR_BASED` and adaptive substepping?

Reply:
I'll add one, also an AD test